service/src/main/java/com/myhome/controllers/UserController.java

```diff
@@ -81,4 +91,34 @@
         .map(response -> ResponseEntity.status(HttpStatus.OK).body(response))
         .orElseGet(() -> ResponseEntity.status(HttpStatus.NOT_FOUND).build());
   }
+
+  @Operation(
+      description = "Request reset user password",
+      responses = {
+          @ApiResponse(
+              responseCode = "200",
+              description = "If password reset request accepted"
+          ),
+          @ApiResponse(
+              responseCode = "400",
+              description = "If password reset request rejected"
+          )
+      })
+  @PostMapping(path = "/user/password",
+      produces = {MediaType.APPLICATION_JSON_VALUE, MediaType.APPLICATION_ATOM_XML_VALUE})
+  public ResponseEntity<Void> userForgotPassword(@RequestParam("action") PasswordActionType action, @RequestBody ForgotPasswordRequest forgotPasswordRequest) {
+    boolean result = false;
+    if (action == PasswordActionType.FORGOT) {
+      result = userService.requestResetPassword(forgotPasswordRequest);
+    } else if (action == PasswordActionType.RESET) {
+      result = userService.resetPassword(forgotPasswordRequest);
+    }
+    if (result) {
+      return ResponseEntity.ok().build();
+    } else {
+      return ResponseEntity.badRequest().build();
+    }
+  }
```

Comment:
May be replace this by two endpoints: `/user/password/forgot` and `/user/password/reset`?

Reply:
Let's leave it as it is for now